metadata-ingestion/src/datahub/ingestion/source/lookml.py

```diff
@@ -429,12 +429,39 @@ def get_including_extends(
     **POSTGRES_TYPES_MAP,
     **SNOWFLAKE_TYPES_MAP,
     "date": DateTypeClass,
+    "date_day_of_month": NumberTypeClass,
+    "date_day_of_week": EnumTypeClass,
+    "date_day_of_week_index": EnumTypeClass,
+    "date_fiscal_month_num": NumberTypeClass,
+    "date_fiscal_quarter": DateTypeClass,
+    "date_fiscal_quarter_of_year": EnumTypeClass,
+    "date_hour": TimeTypeClass,
+    "date_hour_of_day": NumberTypeClass,
+    "date_month": DateTypeClass,
+    "date_month_num": NumberTypeClass,
+    "date_month_name": EnumTypeClass,
+    "date_quarter": DateTypeClass,
+    "date_quarter_of_year": EnumTypeClass,
     "date_time": TimeTypeClass,
+    "date_time_of_day": TimeTypeClass,
+    "date_microsecond": TimeTypeClass,
     "date_millisecond": TimeTypeClass,
     "date_minute": TimeTypeClass,
     "date_raw": TimeTypeClass,
+    "date_second": TimeTypeClass,
     "date_week": TimeTypeClass,
+    "date_year": TimeTypeClass,
+    "date_day_of_year": NumberTypeClass,
+    "date_week_of_year": NumberTypeClass,
+    "date_fiscal_year": TimeTypeClass,
     "duration_day": TimeTypeClass,
+    "duration_hour": TimeTypeClass,
+    "duration_minute": TimeTypeClass,
+    "duration_month": TimeTypeClass,
+    "duration_quarter": TimeTypeClass,
+    "duration_second": TimeTypeClass,
+    "duration_week": TimeTypeClass,
+    "duration_year": TimeTypeClass,
```

Comment:
@jameslamb these types are mainly used for selecting the icons in the UI, and hovering over the icon should still display the raw type. 

I think these choices seem reasonable. I'd weakly prefer using TimeTypeClass over NumberTypeClass/Enum for anything time-related, since it'd be a bit more clear in the UI, but I'll leave the decisions to you.

Reply:
Please make this a number, if that's what the column will contain. Plus, this is more like a time window size as opposed to a particular timestamp.

Reply:
Ah yeah I think I got confused between `duration_year` and `date_year` working through these.

According to https://docs.looker.com/reference/field-reference/dimension-type-reference#available_duration_types, these `duration_*` will all be strings like this:

> 2 years
> 6 weeks

So maybe they should actually be `StringType`?

Reply:
@jameslamb : `StringType` seems reasonable for these duration fields.

Reply:
Ok thanks. I updated all those `duration_*` fields to `StringType` in https://github.com/linkedin/datahub/pull/2972/commits/4241dfdd2faf79b407a778bda43a765b75cfdfe8.